metomi/rose/date.py

```diff
@@ -427,5 +430,56 @@ def _convert_duration(date_time_oper, opts, args):
         sys.exit(1)
 
 
+def upgrade_offset(offset: str) -> str:
+    """Convert offset values in the legacy format
+
+    Args:
+        offset: offset matching [0-9]+[wdhms]
+
+    Returns: Offset in isodate compatible format.
+
+    URL:
+        https://github.com/metomi/rose/issues/2577
+
+    Examples:
+        >>> upgrade_offset('1w')
+        [WARN] This offset syntax is deprecated.
+        'P7DT0H0M0S'
+        >>> upgrade_offset('1w1d1h')
+        [WARN] This offset syntax is deprecated.
+        'P8DT1H0M0S'
+        >>> upgrade_offset('1h1d')
+        [WARN] This offset syntax is deprecated.
+        'P1DT1H0M0S'
+    """
+    Reporter().report(
+        'This offset syntax is deprecated.',
+        prefix=Reporter.PREFIX_WARN, level=Reporter.WARN
+    )
+    sign = '-' if offset[0] == '-' else ''
+    offsets = LEGACY_OFFSET.findall(offset)
+    offsets = {i.upper(): j for j, i in offsets}
+
+    # Rose 2019 did not make any distinction between 1s1m and 1m1s,
+    # so we do an implicit sort here:
+    weeks, days, hours, minutes, seconds = [0 for _ in range(5)]
+    for unit, value in offsets.items():
+        if unit == 'W':
+            weeks = int(value)
+        if unit == 'D':
+            days = int(value)
+        if unit == 'H':
+            hours = int(value)
+        if unit == 'M':
+            minutes = int(value)
+        if unit == 'S':
+            seconds = int(value)
+
+    # Week is not a built-in type:
```

Comment:
This comment is misleading; week is a built-in type in isodatetime, but it can only be used by itself (e.g. `P2W`).
```suggestion
```